### DISCUSSION TITLE
Error building site: POSTCSS: failed to transform "css/eureka.css" (text/css): resource

### DISCUSSION THREAD

**Discussion by Flashpok** [wangchucheng/hugo-eureka#185] [Q&A] [answered]
I have installed Eureka as submodule in Hugo 0.96. It's OK when I try it in local (http://localhost:1313/) But, I can't load it correctly with GITLAB (it does not seem to link to CSS or good url in links). This is the error message when I try to daploy it :

`Error building site: POSTCSS: failed to transform "css/eureka.css" (text/css): resource `

Can you help me ?
Thank you,
C.

**Comment by wangchucheng** [comment]
Hi @Flashpok,

Can you share your repo with us so maybe we can find out the issue you met. This seems not a bug to fix but more like a config issue.

**Reply by Flashpok** [reply]
Of course : https://gitlab.com/hugo1111/eureka01  Thank you for your help !!

**Reply by Flashpok** [reply]
When I want to change baseURL, I have an error if I pu this one "baseURL: "https://hugo1111.gitlab.io/eureka01/". But it seems correct. I don't understand.

**Reply by Flashpok** [reply]
No error if the value is "/" but theme doesn't work.

**Reply by Flashpok** [reply]
I tried : 
baseURL: "https://hugo1111.gitlab.io/eureka01"
and :  
baseURL: "https://hugo1111.gitlab.io/eureka01/"

**Reply by Flashpok** [reply]
When I push "config/_default/config.yaml". 
Results : FAILED JOB : 
Message : 
 `Executing "step_script" stage of the job script
Using docker image sha256:74e25f72f73bfc76f026d510e24bea6bbb757b4839006fb79ba4c3a086805c74 for registry.gitlab.com/pages/hugo/hugo_extended:latest with digest registry.gitlab.com/pages/hugo/hugo_extended@sha256:d1aec7997346d41e61ff912e6a919ea9e118e94a7626f56e421209fb17aa8f9a ...
$ hugo
Start building sites … 
hugo v0.97.0-c07f3626e7c8160943591f4d209977efa02c3dca+extended linux/amd64 BuildDate=2022-04-14T08:45:07Z VendorInfo=gohugoio
Error: Error building site: POSTCSS: failed to transform "css/eureka.css" (text/css). Check your PostCSS installation; install with "npm install postcss-cli". See https://gohugo.io/hugo-pipes/postcss/: this feature is not available in your current Hugo version, see https://goo.gl/YMrWcn for more information
Total in 315 ms
Cleaning up project directory and file based variables
ERROR: Job failed: exit code 255`

**Reply by Flashpok** [reply]
And I don't know how force installation of ""npm install postcss-cli" (I think it's included in hugo-extented ?)

**Reply by wangchucheng** [reply]
I have not checked your repo but one thing for sure is that you do not need to install postcss-cli if you did not modify eureka css.

**Reply by ToddBradley** [reply]
> one thing for sure is that you do not need to install postcss-cli if you did not modify eureka css

Thank you for that information. I have been trying to troubleshoot the same problem as above - `POSTCSS: failed to transform "css/eureka.css"`. And I've now found 3 different possible root causes with ways to fix it. They all result in this same error, so there's no way to know from the error message exactly what hugo is trying to do. One of the fixes I found someone mention was to install `postcss-cli`, but that didn't help me. So it's good to know that it's not necessary in this case.

**Comment by wangchucheng** [comment, marked as answer]
I checked your repo just now and I found that you ignored this:

```
build:
  useResourceCacheWhen: always
```

If you did not modify eureka css, you could just add this. And please refer to the [example config](https://github.com/wangchucheng/hugo-eureka/tree/master/exampleSite/config/_default) to make sure you do not miss anything you want.

**Reply by Flashpok** [reply]
Thank you for your help.
I add : 
`build:
  useResourceCacheWhen: always`

But no change. Gitlab refuse to make the pipeline. (it works only if I set baseURL to "/", but without theme). Same error with good url : https://hugo1111.gitlab.io/eureka11/ (I migrate git for testing) : 

`hugo v0.97.0-c07f3626e7c8160943591f4d209977efa02c3dca+extended linux/amd64 BuildDate=2022-04-14T08:45:07Z VendorInfo=gohugoio
Error: Error building site: POSTCSS: failed to transform "css/eureka.css" (text/css): resource "css/eureka11/css/eureka.css_99793e29f614609ef5e75d384789a0c9" not found in file cache`

I don't understand (No problem with other themes)

**Reply by wangchucheng** [reply]
This error message you just posted is indeed an issue when you set a path instead of a domain in baseURL, I have opened an issue #186 and will look at it later. Should be easy to fix, thanks!

**Reply by Flashpok** [reply]
thank you for your help ! (link to the git: https://gitlab.com/hugo1111/eureka11/)
